x-pack/elastic-agent/pkg/agent/cmd/enroll.go

```diff
@@ -308,6 +311,24 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command, args []string) error {
 
 	ctx := handleSignal(context.Background())
 
+	if localFleetServer := fServer != ""; localFleetServer {
+		if fInternalHost == "" {
```

Comment:
What is the use case where we would not use localhost? Should we even support something else?

Reply:
Fleet-server and its own Elastic Agent are always gonna be on the same host aren't they?

Reply:
It must be on the same host. Elastic Agent starts fleet-server as a process.

Reply:
Then it will always be a localhost communication. Gotcha.

Reply:
Should always be on localhost, however making it overridable could be useful for testing.